Update custom loading doc to include spectral_axis in Spectrum1D object (#641)
astropy/specutils: docs/custom_loading.rst

    import os

    from astropy.io import fits
    from astropy.nddata import StdDevUncertainty
    from astropy.table import Table
    from astropy.units import Unit
    from astropy.wcs import WCS

    from specutils.io.registers import data_loader
    from specutils import Spectrum1D

    # Define an optional identifier. If made specific enough, this circumvents the
    # need to add ``format="my-format"`` in the ``Spectrum1D.read`` call.
    def identify_generic_fits(origin, *args, **kwargs):
        return (isinstance(args[0], str) and
                os.path.splitext(args[0].lower())[1] == '.fits')

    @data_loader("my-format", identifier=identify_generic_fits,
                 extensions=['fits'])
    def generic_fits(file_name, **kwargs):
        with fits.open(file_name, **kwargs) as hdulist:

            tab = Table.read(file_name)

            uncertainty = StdDevUncertainty(tab["err"])
            lamb = tab["spectral_axis"] * Unit('')
            data = tab["flux"] * Unit('')

        return Spectrum1D(flux=data, spectral_axis=lamb, uncertainty=uncertainty)

The above example makes the assumption that the units of the wavelength and the flux are

> stored in the table's ``spectral_axis`` and ``flux`` columns as ``Quantity`` objects.

Reviewer: I'm confused by this statement, as `lamb` and `data` are both being made into `Quantity` objects using explicitly empty `Unit` instances. How are we assuming that the units of the wavelength and flux are in the columns?

            meta = {'header': header}
            wcs = WCS(hdulist[0].header)
            uncertainty = StdDevUncertainty(tab["err"])

>             lamb = tab["spectral_axis"] * Unit("Angstrom")

Reviewer 2: ```suggestion
            lamb = Quantity(tab["spectral_axis"])
```

>             lamb = tab["spectral_axis"] * Unit("Angstrom")

Reviewer 2: ```suggestion
            lamb = Quantity(tab["spectral_axis"])
```

>             data = tab["flux"] * Unit("Jy")

Reviewer 2: ```suggestion
            data = Quantity(tab["flux"])
```

>             data = tab["flux"] * Unit("Jy")

Reviewer 2: ```suggestion
            data = Quantity(tab["flux"])
```

>     @custom_writer("fits-writer")

Reviewer 2: ```suggestion
    @custom_writer("my-format")
```

>     def generic_fits(spectrum, file_name, **kwargs):

Reviewer 2: ```suggestion
    def my_format_writer(spectrum, file_name, **kwargs):
```

        flux = spectrum.flux
        disp = spectrum.spectral_axis
        meta = spectrum.meta

        tab = Table([disp, flux], names=("spectral_axis", "flux"), meta=meta)

        tab.write(file_name, format="fits")

> Please note that it is assumed that the axis (of the ``flux`` and ``spectral_axis``)

Reviewer 2: ```suggestion
Note that this will store the units of the ``flux`` and ``spectral_axis`` as attributes of the ``Quantity`` columns in the ``Table``, which is saving them automatically to the FITS file.
```

> units information is stored in ``Table`` object as introduced in the definition

Reviewer 2: Sorry, it was not clear to me what you meant by "introduced in the definition" here; anyway this is not assuming anything, since these two `Spectrum1D` attributes _must_ have `unit` properties, and a `Table` will preserve them as `Quantity` columns.
Also added my suggestions for consistent naming above (see my comment in the main discussion); I had not noticed that this section was using yet different names (and also duplicating the function definition of `generic_fits` if followed verbatim).

Author: Sure, have incorporated your suggestion for consistent naming.

        tab.write(file_name, format="fits")

Please note that it is assumed that the axis (of the ``flux`` and ``spectral_axis``)
units information is stored in the ``meta`` parameter as introduced in the definition
of ``generic-fits`` as the ``custom_writer()`` above.

> 

Reviewer 2: This is not correct. In the general case we should not make any assumptions on the information stored in the `meta` attribute of a `Spectrum1D` object. However the units _are_ guaranteed to be already part of `flux` and `spectral_axis`, because the are verified to be `Quantity`instances on initialisation. That's why just
```
        flux = spectrum.flux
        disp = spectrum.spectral_axis
```
should be used.

Author: I have changed the wording to match your descriptions.
